pkg/apis/externaldns/types.go

```diff
@@ -71,5 +74,9 @@ func (cfg *Config) ParseFlags(args []string) error {
 	flags.BoolVar(&cfg.DryRun, "dry-run", true, "dry-run mode")
 	flags.BoolVar(&cfg.Debug, "debug", false, "debug mode")
 	flags.BoolVar(&cfg.Version, "version", false, "display the version")
+	flags.StringVar(&cfg.Registry, "registry", "noop", "type of registry for ownership")
```

Comment:
As long as you have a limited list of providers, I would put it in the docstring. Maybe <noop|txt> is enough to add